### PR TITLE
changing search bar controls and adding separate menu functionality

### PR DESCRIPTION
**Pull request by clairepo**
- auto highlight text on search bar focus
- when focus show search control, after submit show cancel control
- functionality to only search dishes from current menu

Back end PR: https://github.com/Nomi-Technologies/menu-service/pull/40

### CODE REVIEW

File: src/components/dashboard/menu-table/menu-table.js [hunk 1]
```diff
@@ -284,16 +287,27 @@ const MenuTable = (props) => {
...
                         <input className='cancelSearch' type='image' alt="Reset search" src={CancelIcon} onClick={(e) => {
                             e.preventDefault();
                             setSearchBoxValue('');
                             setIsSearching(false);
-                        }}/> : null
+                            setAfterSubmit(false);
+                        }}/> : 
+                        <input className='submitSearch' type='image' alt="Submit" src={SearchIcon} onClick={(e) => {
+                            setAfterSubmit(true);
+                        }}/>
```
Comment (Tony1023) [inline src/components/dashboard/menu-table/menu-table.js:309]: Hmmm I remember you could exit searching mode when you hit enter with an empty search box. How does someone exit search now when they clear the search box by hitting backspace? I'm also having trouble following afterSubmit and isSearching...

Comment (clairepo) [reply src/components/dashboard/menu-table/menu-table.js:309]: Sorry so many variables, let me try to explain.

The customer can exit search by either pressing enter or magnifying glass icon with empty text box OR by pressing X icon while there's text in the box.

isSearching - changes to true after the search is submitted the first time, user can continue searching and isSearching doesn't turn back to false until after they have exited the search (X icon or submitting with empty text box). This variable is used to tell if search results table should be displayed or normal menu should be displated.

afterSubmit - changes to true after the search is submitted, changes back to false when the user enters the text box again to make another search. This variable is used to tell if we should display magnifying glass icon or X icon (afterSubmit = true then show X icon, afterSubmit = false show magnifying glass icon)

These two only differ when a user wants to execute multiple searches in a row without exiting the search in between. Let me know if that makes sense and if I answered your question.

Comment (clairepo) [reply src/components/dashboard/menu-table/menu-table.js:309]: Looks like when the user presses submit with empty box doesn't work anymore to exit the search, I will fix this

Comment (clairepo) [reply src/components/dashboard/menu-table/menu-table.js:309]: Also, writing this explanation made me realize I could just use a combination of isSearching and searchBoxFocused and get rid of afterSubmit (:

**Pull request merged**